src/commands/base.ts

```diff
@@ -224,4 +225,84 @@ export default abstract class Base extends Command {
     hasToken(): boolean {
         return this.authToken !== ''
     }
+
+    public async populateParameters<ResourceType>(
+        paramClass: ClassConstructor<ResourceType>,
+        prompts: Prompt[],
+        flags: Record<string, unknown> = {},
+        isUpdate = false,
+    ): Promise<ResourceType> {
+        if (flags.headless) {
+            const params = plainToClass(paramClass, flags)
+            const errors = validateSync(params as Record<string, unknown>, {
+                whitelist: true,
+                skipMissingProperties: isUpdate,
+            })
+            reportValidationErrors(paramClass.name, errors)
+            return params
+        }
+        const filteredPrompts = this.filterPrompts(prompts, flags)
+        const answers = await this.populateParametersWithInquirer(filteredPrompts, isUpdate)
+
+        const params = plainToClass(paramClass, this.mergeFlagsAndAnswers(flags, answers))
+        const errors = validateSync(params as Record<string, unknown>, {
+            whitelist: true,
+            skipMissingProperties: isUpdate,
+        })
+        reportValidationErrors(paramClass.name, errors)
+        return params
+    }
+
+    // Filter out prompts that already have values provided by flags
+    private filterPrompts(prompts: Prompt[], flags: Record<string, unknown>) {
+        let filteredPrompts: Prompt[] = []
+        Object.keys(flags).forEach((key) => {
+            if (flags[key]) {
+                filteredPrompts = prompts.filter((prompt) => prompt.name !== key)
+            }
+        })
+        return filteredPrompts
+    }
+
+    // Merge answers from prompts with values provided by flags
+    private mergeFlagsAndAnswers(flags: Record<string, unknown>, answers: Record<string, unknown>) {
+        const updatedFlags = { ...flags }
+        Object.keys(updatedFlags).forEach((key) => {
+            if (updatedFlags[key] === undefined) {
+                updatedFlags[key] = answers[key]
+            }
+        })
+        return updatedFlags
+    }
+
+    private async populateParametersWithInquirer(prompts: Prompt[], isUpdate = false) {
+        let filteredPrompts = prompts
+        if (isUpdate) {
```

Comment:
rather than passing this parameter around, it could make sense to override this method in the UpdateCommand class that extends the Base class